pandas/tests/util/test_assert_series_equal.py

```diff
@@ -456,3 +452,12 @@ def test_large_unequal_ints(dtype):
     right = Series([1577840521123543], dtype=dtype)
     with pytest.raises(AssertionError, match="Series are different"):
         tm.assert_series_equal(left, right)
+
+
+@pytest.mark.parametrize("check_exact", [True, False])
+@pytest.mark.parametrize("val", [3, 3.5])
+def test_ea_and_numpy_no_dtype_check(val, check_exact):
+    # GH#56651
+    left = Series([1, 2, val])
```

Comment:
Could you also parameterize where this is `dtype=object` too? I think that should pass this test too?

Reply:
Done, and yes this works too